classes/xml/XMLComment.inc.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/**
+ * @file classes/xml/XMLComment.inc.php
+ *
+ * Copyright (c) 2013-2015 Simon Fraser University Library
+ * Copyright (c) 2000-2015 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class XMLComment
+ * @ingroup xml
+ *
+ * @brief Extension of XMLNode for a simple DOM-style comment.
+ */
+
+import ('lib.pkp.classes.xml.XMLNode');
+
+class XMLComment {
```

Comment:
Have you considered making this extend XMLNode? In DOM implementations, OTOH, comments are typically nodes; the way this is implemented there's an unstated assumption that both XMLNode and XMLComment will behave similarly. An interface or class extension would make that explicit.

Note that this whole set of classes will be removed (with great pleasure) in favour of the standard PHP5 toolset at some point. But that won't be done for OJS 2.x.